libs/vertexai/langchain_google_vertexai/tools/vertex_search.py

```diff
@@ -0,0 +1,103 @@
+from typing import Any, Dict, Optional
+
+from langchain_core.pydantic_v1 import Field
+from langchain_core.tools import BaseTool
+
+
+class VertexSearchTool(BaseTool):
+    """Class that exposes a tool to interface with an App in Vertex Search and
+    Conversation. Now Agent Builder.
+    """
+
+    project_id: str
+    """ Id of the GCP project the App is in"""
+
+    engine_id: str
+    """ Id of the Vertex Search App"""
+
+    location: str = "global"
+    """ Location of the Vertex Search App. Defaults to `global`"""
+
+    return_snippet: bool = True
+    """ Whether to return a snippet of the search results"""
+
+    max_number_of_documents: int = 10
+    """ Maximum number of documents to retrieve in the search"""
+
+    summary_result_count: int = 3
+    """ Number of documents to include in the summary"""
+
+    summary_include_citations: bool = True
+    """ Whether to include citations in the summary """
+
+    summary_prompt: Optional[str] = None
+
+    summary_spec_kwargs: Dict[str, Any] = Field(default_factory=dict)
+    """ Additional keyword arguments for 
+        `discoveryengine.SearchRequest.ContentSearchSpec.SummarySpec`
+    """
+    search_request_kwargs: Dict[str, Any] = Field(default_factory=dict)
+    """ Additional keyword arguments for 
+        `discoveryengine.SearchRequest`
+    """
+
+    content_search_spec_kwargs: Dict[str, Any] = Field(default_factory=dict)
+    """ Additional keyword arguments for 
+        `discoveryengine.ContentSearchSpec`
+    """
+
+    def _run(self, search_query: str, **kwargs: Any) -> str:
+        """Runs the tool.
+
+        Args:
+            search_query: The query to run by the agent.
+
+        Returns:
+            The response from the agent.
+        """
+
+        import google.cloud.discoveryengine_v1 as discoveryengine  # type: ignore[import-untyped, unused-ignore]
```

Comment:
should it be at langchain_google_community maybe? where we already have an integration for Vertex Search?